components/elm/cime_config/buildnml

```diff
@@ -181,7 +181,10 @@ def buildnml(case, caseroot, compname):
         if elm_nml_use_case != "UNSET": usecase = "-use_case {}".format(elm_nml_use_case)
 
         start_ymd = run_startdate.replace("-", "")
-        ignore = "-ignore_ic_year" if ("-01-01" in run_startdate or "-09-01" in run_startdate) else "-ignore_ic_date"
+
+        # change to always ignore_ic_year to allow more flexibility in specifying IC for different start date of year
+        # ignore = "-ignore_ic_year" if ("-01-01" in run_startdate or "-09-01" in run_startdate) else "-ignore_ic_date"
+        ignore = "-ignore_ic_year"
```

Comment:
I think this is fine for now, but eventually we will have a case for the same month/day but a different year and we'll be confused why it isn't working right. Does "101" get expanded to "year=00"? If so, we could ignore_ic_year only when year=00 to avoid future problems...

Reply:
True, Peter. If we come to that one day, we need to make it even more flexible, with modifications inside build-namelist that this python script invokes. In the original script, it goes with either 'ignore_ic_year' or 'ignore_ic_date', not suitable for our current purpose. Also don't see this change to allow more flexibility will break any existing tests.

Reply:
BTW, good thought about the ignore option for year=00 only. Changes inside the perl build-namelist would still be needed for make it work. I did test before creating this PR by ignoring none. It did not work out for all our land IC options we plan to run (e.g,  DYAMOND-1, DYAMOND-2, and the four seasons for a year).